### ISSUE TITLE
Add ReadJournal support for PersistenceQuery

### ISSUE THREAD

**Issue by kaleho** [petabridge/Akka.Persistence.Azure#42]
I am in the process of creating the AzureTableStorageReadJournal. I have modeled most of it after the MongoDB journal. I can see the the AzureTableStorageJournal does not support Tags and I would like to provide that. Since Table Storage does not have an array property type I was wondering what the best approach for that would be since it would also need to be queried?

**Comment by Aaronontheweb** [comment]
@kaleho we're going to need to add a metadata table of some kind in order to handle this, so it'll likely need to be done through schema...

**Comment by kaleho** [comment]
@Aaronontheweb ya, as I got into it I gathered that there are several things like that. Been looking to the community Redis plugin for inspiration. At this point, given the requirements for supporting the aforementioned tag querying, the mechanics of Azure table storage and how support for AllPersistenceIds would work I am in need of more direction. My plan so far was to add another table with a "journal:persistenceIds" as the partitionkey, a padded incremented number for the rowkey, a column for the persistenceId and one for each tag. That should satisfy the need to keep the records in the same partition to keep it performant and keep the tag query down to a simple equals. Is that reasonable? Sorry, I am happy to do the work but also realize I am in need of more guidance.

**Comment by kaleho** [comment]
@Aaronontheweb So, I have a completed implementation at the following branch of my fork: https://github.com/kaleho/Akka.Persistence.Azure/tree/completed-implementation

Regrettably, I had the work done and then I read your blog post on using GitHub professionally and I was about to make a pretty monolithic pull request. So I stopped and started to piece-meal my changes in as small PRs but that, along with the two tests that I describe below has gotten me pretty drained.

There are two tests that will not pass if the `AzureTableCurrentEventsByPersistenceIdSpec` is run in its entirety but will pass if run individually. 

- `ReadJournal_CurrentEventsByPersistenceId_should_return_empty_stream_for_cleaned_journal_from_0_to_MaxLong`
- `ReadJournal_CurrentEventsByPersistenceId_should_return_remaining_values_after_partial_journal_cleanup`

I need to clean it up some, there are some properties in the table entities that need to get cleaned up but beyond that I would consider it pretty close. Also need to clean up some LINQ usage.

All the design decisions I made were with performance in mind and trying to be sympathetic with my understanding of how Azure Table Storage works and Microsoft's prescriptive guidance on the subject.

I know you are a busy guy but I would like to know where to go from here/get some feedback. Hopefully it is not too far off.
